### PR TITLE
Update OLM version to 0.14.1

### PR DESCRIPTION
**Pull request by baijum**
### Motivation

This was created to make SDK update a focused one.
Refer PR #469

### Changes

Steps:

1. Update go.mod based on the dependencies from go.mod in OLM:
   https://github.com/operator-framework/operator-lifecycle-manager/tree/0.14.1
2. Run: `go mod tidy`
3. Run: `go mod vendor`
4. Changed `test/mocks/mocks.go` file based on API changes

### Testing

All tests should pass.

For further more details refer the CONTRIBUTING.md

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -35,26 +34,36 @@ require (
 // Pinned to kubernetes-1.16.2
```
Comment (isutton) [inline go.mod:34]: Can you validate whether this is still true? `k8s.io/kubernetes` has been removed from `require`, and `k8s.io/api` and others have been changed to `0.17.1`.

File: go.mod [hunk 2]
```diff
@@ -12,19 +12,18 @@ require (
 	github.com/openshift/api v3.9.1-0.20190924102528-32369d4db2ad+incompatible
 	github.com/openshift/custom-resource-status v0.0.0-20190822192428-e62f2f3b79f3
 	github.com/operator-backing-service-samples/postgresql-operator v0.0.0-20191023140509-5c3697ed3069
-	github.com/operator-framework/operator-lifecycle-manager v0.0.0-20191115003340-16619cd27fa5
+	github.com/operator-framework/operator-lifecycle-manager v0.0.0-20200130164400-12c06cfc05c4
```
Comment (isutton) [inline go.mod:15]: Any specific reason `v0.14.1` can't be used as a version there?

File: go.mod (hunk 1 above)
Comment (baijum) [reply go.mod:34]: `k8s.io/kubernetes` is an indirect dependency that needs not be there in the "require" section.
All the tests are passing. Not sure what another validation is required.

File: go.mod [hunk 3]
```diff
@@ -35,26 +34,36 @@ require (
...
-	k8s.io/cloud-provider => k8s.io/cloud-provider v0.0.0-20191016115326-20453efc2458
-	k8s.io/cluster-bootstrap => k8s.io/cluster-bootstrap v0.0.0-20191016115129-c07a134afb42
-	k8s.io/code-generator => k8s.io/code-generator v0.0.0-20191004115455-8e001e5d1894
-	k8s.io/component-base => k8s.io/component-base v0.0.0-20191016111319-039242c015a9
+
+	github.com/docker/docker => github.com/moby/moby v0.7.3-0.20190826074503-38ab9da00309 // Required by Helm
+	github.com/openshift/api => github.com/openshift/api v3.9.1-0.20190924102528-32369d4db2ad+incompatible
+	github.com/openshift/client-go => github.com/openshift/client-go v0.0.0-20190923180330-3b6373338c9b
+	github.com/prometheus/client_golang => github.com/prometheus/client_golang v0.9.3-0.20190127221311-3c4408c8b829
+	helm.sh/helm/v3 => helm.sh/helm/v3 v3.0.0-beta.5.0.20200123114618-5e3c7d7eb86a
```
Comment (isutton) [inline go.mod:42]: Is `helm.sh/helm/v3` a transient or direct dependency?

File: go.mod (hunk 2 above)
Comment (baijum) [reply go.mod:15]: `go mod tidy` did this change.

File: go.mod (hunk 3 above)
Comment (baijum) [reply go.mod:42]: It's a transient dependency. See this: https://github.com/operator-framework/operator-lifecycle-manager/blob/0.14.1/go.mod

File: go.mod (hunk 2 above)
Comment (baijum) [reply go.mod:15]: For more details read this FAQ: https://github.com/golang/go/wiki/Modules#can-a-module-consume-a-package-that-has-not-opted-in-to-modules

Comment (baijum) [reply go.mod:15]: Raised an issue in OLM project: https://github.com/operator-framework/operator-lifecycle-manager/issues/1524

**Pull request merged**